src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,60 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 0) {
+            return new int[0];
+        }
+        if (matrix.length == 1) {
+            return matrix[0];
+        }
+        int[] resultArray = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            resultArray[i] = matrix[i][i];
+        }
+        return resultArray;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 0) {
+            return new int[0];
+        }
+        if (matrix.length == 1) {
+            return matrix[0];
+        }
+        int[] resultArray = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            resultArray[i] = matrix[i][matrix.length - i - 1];
+        }
+        return resultArray;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        if (matrix.length == 0) {
+            return 0;
+        }
+        int result = Integer.MAX_VALUE;
```

Comment:
you don't need constant here, you may just choose assign result to first matrix element

Reply:
Fixed as well.